src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs

```diff
@@ -523,18 +523,18 @@ internal bool SetInvalidStringFailure()
 
             internal bool SetArgumentNullFailure(string argumentName)
             {
-                if (!_throwOnFailure)
+                if (_throwOnFailure)
                 {
-                    return false;
+                    Debug.Assert(argumentName != null);
+                    ArgumentNullException.Throw(argumentName);
                 }
 
-                Debug.Assert(argumentName != null);
-                throw new ArgumentNullException(argumentName, SR.ArgumentNull_String);
+                return false;
             }
 
             internal bool SetOverflowFailure()
             {
-                if (!_throwOnFailure)
+                if (_throwOnFailure)
```

Comment:
This one is inverted

Reply:
Oops... not sure how that happened. Thanks.